### ISSUE TITLE
[REGRESSION]: session storage failing >= 1.25.0 in firefox

### ISSUE THREAD

**Issue by rainerhahnekamp** [microsoft/playwright#16937]
**Context:**
- GOOD Playwright Version: 1.24
- BAD Playwright Version: 1.25
- Operating System: Mac, Window (didn't test on Linux)

**Code Snippet**

```typescript
// ./global-setup


import { chromium, FullConfig } from "@playwright/test";

async function globalSetup(config: FullConfig) {
  const browser = await chromium.launch();
  const page = await browser.newPage();
  await page.goto("https://genuine-narwhal-f0f8ad.netlify.app/");
  await page.locator("data-testid=btn-sign-in").click();
  await page.locator("input[name=email]").fill("john.list@host.com");
  await page.locator("input[name=password]").fill("John List");
  await page.locator("button[type=submit]").click();
  await page
    .locator("data-testid=p-username", { hasText: "John List" })
    .waitFor();

  await page.context().storageState({ path: "john-list.json" });
  await page.close();
}

export default globalSetup;
```

```typescript
// ./tests/advanced-holidays.spec.ts

import { test, expect } from "@playwright/test";

test.use({ storageState: "john-list.json" });

test.describe("Advanced Holidays", () => {
  test("sign-in test", async ({ page }) => {
    await page.goto("");

    await expect(page.locator("data-testid=p-username")).toHaveText(
      "Welcome John List"
    );
  });
});

```

**Describe the bug**

The session storage feature does not work on firefox anymore. starting with version 1.25. You can also clone the project from https://github.com/rainerhahnekamp/playwright-heise-dev

**Comment by mxschmitt** [comment]
I can repro on 1.25 and ToT with Firefox and WebKit. This might be related to some magic which auth0 does in the background, we few related issues in the past.

**Comment by rainerhahnekamp** [comment]
I see. Do you think you can fix that?

**Comment by rainerhahnekamp** [comment]
Thanks a lot, will try and let you, ok?

**Comment by pavelfeldman** [comment]
We don't have a nice doc for that, so I'm leaving it open for the doc. My recommendation would be to create 3 playwright projects that require logged-in state in `playwright.config.ts`. You would then specify different `storageState` values for these projects in that config. That way your global setup runs, it logs in in 3 browsers, then projects run and they pick up respective auth. We are going to have another feature in v1.26 that would make it much more straightforward, but for now that's the only nicer way I can think of.

**Comment by ScottGuymer** [comment]
This is also an issue when you want to clear the storageState for a test

``` 
test.use({ storageState: undefined });

test.describe("test", () => {
  test("test without state", async ({ page }) => {
    // some test without using state.
  });
});
``

This works in 1.24.1 but not in 1.25.1

**Comment by mxschmitt** [comment]
@ScottGuymer, your issue is most likely different, please file a separate one with a repro and more context.

**Comment by mxschmitt** [comment]
Moved the WebKit issue into https://github.com/microsoft/playwright/issues/17047.

Thanks for reporting @rainerhahnekamp!

**Comment by rainerhahnekamp** [comment]
@pavelfeldman, @mxschmitt: I tried the proposed approach but without success. Maybe I've missed something? You can see my changes in https://github.com/rainerhahnekamp/playwright-heise-dev/commit/f5748106f60bd95acda7619e7221e86f66366b4b

Thank you!

**Comment by mxschmitt** [comment]
> @pavelfeldman, @mxschmitt: I tried the proposed approach but without success. Maybe I've missed something? You can see my changes in https://github.com/rainerhahnekamp/playwright-heise-dev/commit/f5748106f60bd95acda7619e7221e86f66366b4b
> 
> Thank you!

It's a bug on our side, the proposed workaround won't work. We'll fix the Firefox issue most likely in 1.25.2 and for WebKit in 1.26.

**Comment by rainerhahnekamp** [comment]
I see, thanks and I am looking forward to 1.26 then. Cheers!

**Comment by aslushnikov** [comment]
This is fixed and will be public in 1.25.2 as well.
